examples/platform/linux/AppMain.cpp

```diff
@@ -315,12 +315,12 @@ void ChipLinuxAppMainLoop()
     initParams.testEventTriggerDelegate = &testEventTriggerDelegate;
 #endif
 
+    // We need to set DeviceInfoProvider before Server::Init to setup the storage of DeviceInfoProvider properly.
+    DeviceLayer::SetDeviceInfoProvider(&gExampleDeviceInfoProvider);
+
     // Init ZCL Data Model and CHIP App Server
     Server::GetInstance().Init(initParams);
 
-    gExampleDeviceInfoProvider.SetStorageDelegate(&chip::Server::GetInstance().GetPersistentStorage());
```

Comment:
> but it does not fix the initialization order issue , we need to first setup DeviceInfoProvider then setup the storage delegate of DeviceInfoProvider.

Why?  Why does the current flow (which looks like it should end up with a provider set to the right object, which is pointing to the right delegate), not work?

Reply:
The current flow works with this workaround, after we adjust the initialization order, the storage get configured in Server::GetInstance().Init(initParams); so we don't need this workaround anymore.

Reply:
I understand why the new code works.  What I am trying to understand is why the current code does not work.  This PR says it's fixing https://github.com/project-chip/connectedhomeip/issues/21798, which suggests that issue is caused by the current code, right?

Reply:
I see, my debug log show the storage has not been set on Linux platform, but I didn't notice we already have the workaround on Linux platform when I registered the issue.

I think the workaround is extra and we don't need this after we fix the ordering issue.

Reply:
Oh, so https://github.com/project-chip/connectedhomeip/issues/21798 was filed based on code inspection, not observing incorrect behavior?

Reply:
Yes, that is out of date, removed.